### ISSUE TITLE
[BUG] Environment variable substitution fails silently if "__format" parser is not supported

### ISSUE THREAD

**Issue by KalleV** [node-config/node-config#753]
**Describe the bug**
Defining a custom-environment-variables.json file with "__format" definitions that do not match a supported parser leads to the configuration being silently ignored.

Example:
```json
{
  hostname: {
    "__name": "HOSTNAME",
    "__format": "string"
  }
}
```

**Expected behavior**
An error when attempting to use an unsupported parser.

**Screenshots**
N/A

### Please tell us about your environment:

 -  node-config version: 3.3.9
 -  node-version: 20.x
 
### Other information

Related code:
 - https://github.com/node-config/node-config/blob/8c52ea28863bdf1562b896ea1f75dada7a0ad11e/lib/config.js#L897-L902
 - https://github.com/node-config/node-config/blob/8c52ea28863bdf1562b896ea1f75dada7a0ad11e/lib/config.js#L1103-L1113

**Comment by lorenwest** [comment]
Agreed it would be an improvement to either output an error message or throw an error if an unknown parser is specified.

**Comment by markstos** [comment]
@KalleV Would you like to contribute a PR for this? Otherwise, the issue is likely to stay open.

**Comment by jdmarshall** [comment]
I'm not sure how I feel about this feature, but this feels like an extension of [strict mode](https://github.com/node-config/node-config/wiki/Strict-Mode)

**Comment by jdmarshall** [comment]
There's a placeholder for this feature in lib/util.js 

https://github.com/node-config/node-config/blob/master/lib/util.js#L1303

**Comment by jdmarshall** [comment]
I've flagged this as a breaking change but it's a relatively small one as these things go. But I still might move it to 6.0
